collect_app/src/main/java/org/odk/collect/android/dao/helpers/InstancesDaoHelper.java

```diff
@@ -74,15 +74,12 @@ public static boolean isInstanceComplete(boolean end, boolean completedByDefault
     // that returns an {@link Instance} object from a path.
     public static Uri getLastInstanceUri(String path) {
         if (path != null) {
-            try (Cursor c = new InstancesDao().getInstancesCursorForFilePath(path)) {
-                if (c != null && c.getCount() > 0) {
-                    // should only be one...
-                    c.moveToFirst();
-                    String id = c.getString(c.getColumnIndex(InstanceColumns._ID));
-                    return Uri.withAppendedPath(InstanceColumns.CONTENT_URI, id);
-                }
+            Instance instance = new DatabaseInstancesRepository().getOneByPath(path);
+            if (instance != null && instance.getStatus().equals(Instance.STATUS_COMPLETE)) {
```

Comment:
Why did you add this additional condition here (`&& instance.getStatus().equals(Instance.STATUS_COMPLETE)`)

Reply:
Hmmm looks like a mistake. I feel like that should probably break something, but it doesn't seem to. I'll look into that.

Reply:
OK I've removed the extra code. Looked like it wasn't causing problems because it was only used in places where the instance would be completed.